safenode/src/protocol/wallet/keys.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use super::error::{Error, Result};
+
+use sn_dbc::MainKey;
+
+use hex::{decode, encode};
+use std::path::Path;
+use tokio::fs;
+
+/// Filename for storing the node's reward (BLS hex-encoded) main key.
+const MAIN_KEY_FILENAME: &str = "main_key";
+/// Filename for storing the node's reward (BLS hex-encoded) public address.
+const PUBLIC_ADDRESS_FILENAME: &str = "public_address";
+
+/// Writes the public address and main key (hex-encoded) to different locations at disk.
+pub(super) async fn store_new_keypair(root_dir: &Path, main_key: &MainKey) -> Result<()> {
+    let secret_key_path = root_dir.join(MAIN_KEY_FILENAME);
```

Comment:
Not bad if we can have a BOM for the file, like: <2 byte version> + <Type of Key>
Type of key for BLS can set to 12381

Reply:
Yeah, secret key handling is a part of the master document in discourse. All that we have done in that area so far has been rather temporary.

If you have an idea for a more long term solution, please do a write up on discourse and add the link to the pinned master document wiki (it has a section about secret key handling).

Reply:
The topic about [key-store](https://maidsafe.discourse.team/t/code-design-topic-keystore-format/758) format in discourse.